drivers/serial/uart_sf32lb.c

```diff
@@ -32,16 +36,71 @@
 #define UART_CR2_STOP_1B FIELD_PREP(USART_CR2_STOP_Msk, 0U)
 #define UART_CR2_STOP_2B FIELD_PREP(USART_CR2_STOP_Msk, 1U)
 
+#define UART_FLAG_IS_SET(flag) \
+	sys_test_bit(UART_REG(UART_ISR), flag)
+
+#define UART_FLAG_CLEAR(flag) \
+	sys_clear_bit(UART_REG(UART_ICR), flag)
+
 /* minimal BRR: INT=1, FRAC=0 (0x10) */
 #define UART_BRR_MIN 0x10U
 
+struct uart_sf32lb_data {
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+	uart_irq_callback_user_data_t irq_callback;
+	void *cb_data;
+#endif
+};
+
 struct uart_sf32lb_config {
 	uintptr_t base;
 	const struct pinctrl_dev_config *pcfg;
 	struct sf32lb_clock_dt_spec clock;
 	struct uart_config uart_cfg;
+#if CONFIG_UART_INTERRUPT_DRIVEN
+	void (*irq_config_func)(const struct device *dev);
+#endif
 };
 
+#if defined(CONFIG_UART_INTERRUPT_DRIVEN)
+static void uart_sf32lb_isr(const struct device *dev)
+{
+	const struct uart_sf32lb_config *config = dev->config;
+	struct uart_sf32lb_data *data = dev->data;
+
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+	if (data->irq_callback) {
+		data->irq_callback(dev, data->cb_data);
+	}
+#endif /* CONFIG_UART_INTERRUPT_DRIVEN */
+	if (UART_FLAG_IS_SET(USART_ISR_ORE_Pos)) {
+		UART_FLAG_CLEAR(USART_ICR_ORECF_Pos);
+	}
```

Comment:
this silently clears overrun flag, please, check uart.h on how to handle certain error conditions properly (probably others below?)